test/index.test.js

```diff
@@ -74,4 +74,20 @@ describe('RadioParser', () => {
       done();
     });
   });
+
+  it('Should properly emit metadata with notifyOnChangeOnly property', done => {
```

Comment:
It will be better to mock the implementation here. So, you mock the metadata, like it is got from a real radio station, subscribe to the `metadata` event and expect that `metadata` was triggered only once.

And another test case where there is two different metadata and you expect that event `metadata` will be fired twice.

Jest has all you need to mock the implementation, expecations, etc... https://jestjs.io/docs/en/expect

Reply:
Here is an example of what I'm talking about - https://github.com/ghaiklor/icecast-parser/blob/master/test/StreamReader.test.js#L10-L31

You can copy paste the test and change the parts of the code there. Add a new flag to the parser, listen for events and make sure that only one event was fired during the test. The same for another, but there emit two different metdata and expect to event be fired twice.

Reply:
In the test, you mock `StreamReader`. I understand that we need to mock `RadioParser`, but I don't understand how to do it. You write to copy and paste the code and add flags, but we cant add flags to `StreamReader`. In all other tests of `RadioParser` you take the real radio.

Reply:
You can mock a stream reader that is used in radio parser. See https://jestjs.io/docs/en/mock-functions#mocking-modules

So, you are mocking a stream reader via jest and configure the mock to emit subsequent metadata events with the same metadata for one test, and different metadata for another test. Since stream reader is mocked and behaves as if a real radio station was listening, the radio parser will think there is a real radio station stuff going on.

Though, you need to be careful, and not to mock it for all the tests, but only for your two.